mbedtls/src/pk/mod.rs

```diff
@@ -319,6 +325,46 @@ impl Pk {
         Ok(ret)
     }
 
+    /// Constructs a private key from elliptic curve components.
+    ///
+    /// This function requires a random number generator (RNG) because it uses the `EcPoint::mul`
+    /// function which requires an RNG for blinding.
+    ///
+    /// It initializes a `Pk` context, generates the public key point by multiplying the curve generator
+    /// with the private key, and sets up the private key context with the generated curve, private key,
+    /// and public point.
+    ///
+    /// # Arguments
+    ///
+    /// * `rng` - The RNG used for blinding in the `EcPoint::mul` function.
+    /// * `curve` - The elliptic curve group to use for key generation.
+    /// * `private_key` - The private key as an MPI (Multiple Precision Integer).
+    ///
+    /// # Returns
+    ///
+    /// * `Result<Pk>` - A `Pk` context filled with the generated private key on success.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if:
+    ///
+    /// * Fails to genearte `EcPoint` from given EcGroup in `curve`.
+    /// * The underlying C `mbedtls_pk_setup` function fails to set up the `Pk` context.
+    /// * The `EcPoint::mul` function fails to generate the public key point.
+    pub fn private_from_ec_components_with_rng<F: Random>(mut curve: EcGroup, private_key: Mpi, rng: &mut F) -> Result<Pk> {
```

Comment:
Should be called `private_from_ec_scalar_with_rng` .

Reply:
Ok I will create another PR for update these